### PR TITLE
Feature/configfromobj support programmatically created credentials and cluster configs

### PR DESCRIPTION
**Pull request by zfengms**
support programmatically created credentials and cluster configs

### CODE REVIEW

File: tests/testthat/test-set-credentials.R [hunk 1]
```diff
@@ -0,0 +1,55 @@
...
+  credentials <- list(
+    "batchAccount" = list(
+      "name" = "batchaccountname",
+      "key" = "batchaccountkey",
+      "url" = "https://batchaccountname.region.batch.azure.com"
+    ),
+    "storageAccount" = list("name" = "storageaccountname",
+                            "key" = "storageaccountkey")
+  )
+  doAzureParallel::setCredentialsObject(credentials)
```
Comment (paselem) [inline tests/testthat/test-set-credentials.R]: Is this different than the setCredentials call? Can't we have an override?

File: tests/testthat/test-set-credentials.R [hunk 2]
```diff
@@ -0,0 +1,55 @@
...
+    "rPackages" = list(
+      "cran" = c(),
+      "github" = c(),
+      "bioconductor" = c(),
+      "githubAuthenticationToken" = c()
+    ),
+    "commandLine" = c()
+  )
+  
+  cluster <- doAzureParallel::makeClusterObject(clusterConfig)
```
Comment (paselem) [inline tests/testthat/test-set-credentials.R]: Same as above regarding overrides.

File: tests/testthat/test-set-credentials.R [hunk 3]
```diff
@@ -0,0 +1,55 @@
+# Run this test for users to make sure the set credentials from R object features
+# of doAzureParallel are still working
+context("set credentials from R object scenario test")
```
Comment (paselem) [inline tests/testthat/test-set-credentials.R]: Is the 'old' way of passing a file still working? Probably worth adding tests to make sure those calls work as before.

File: docs/32-programmatically-generate-config.md [hunk 4]
```diff
@@ -0,0 +1,56 @@
...
+        "max" = 1
+      ),
+      "autoscaleFormula" = "QUEUE"
+    ),
+    "containerImage" = "rocker/tidyverse:latest",
+    "rPackages" = list(
+      "cran" = c(),
+      "github" = c(),
+      "bioconductor" = c(),
+      "githubAuthenticationToken" = c()
```
Comment (paselem) [inline docs/32-programmatically-generate-config.md]: This should no longer be here. It belongs in the credentials object.

File: R/cluster.R [hunk 5]
```diff
@@ -395,6 +419,16 @@ setCredentials <- function(fileName = "az_config.json") {
   print("Your azure credentials have been set.")
 }
 
+#' Set azure credentials to R session from specified R object.
+#'
+#' @param credentialsConfig The credentials R object was created by user programatically
+#'
+#' @export
+setCredentialsObject <- function(credentialsConfig) {
```
[the remark this answers is not in the document]
Comment (zfengms) [reply R/cluster.R]: put credentials related methods in its own R file

File: tests/testthat/test-set-credentials.R (hunk 1 above)
Comment (zfengms) [reply tests/testthat/test-set-credentials.R]: combined both methods to one setCredentials() function and use class() to check whether it's a json file or a R credentials object.

File: docs/33-programmatically-generate-config.md [hunk 6]
```diff
@@ -0,0 +1,56 @@
...
+        "max" = 1
+      ),
+      "autoscaleFormula" = "QUEUE"
+    ),
+    "containerImage" = "rocker/tidyverse:latest",
+    "rPackages" = list(
+      "cran" = list(),
+      "github" = list(),
+      "bioconductor" = list(),
+      "githubAuthenticationToken" = ""
```
Comment (paselem) [inline docs/33-programmatically-generate-config.md]: remove this as we should no longer have this in the cluster config. It should be part of the credentials config.

**Pull request merged**